src/components/dropdown/react/Dropdown.tsx

```diff
@@ -125,31 +125,33 @@ const Dropdown: React.FC<DropdownProps> = ({
         </div>
     );
 
+    const onEscapeHandler = onClose ? onEscapePressed(onClose) : undefined;
+
     useEffect(() => {
-        if (!onClose) {
+        if (!onEscapeHandler) {
             return undefined;
         }
         if (closeOnEscape && showDropdown && wrapperRef.current) {
-            window.addEventListener('keydown', onEscapePressed(onClose));
+            window.addEventListener('keydown', onEscapeHandler);
         } else {
```

Comment:
The else branch is not necessary. The useEffect will automatically call your `removeEventListener` inside your clean up lambda if the dependencies change